Examples/Io/Root/include/ActsExamples/Io/Root/RootMeasurementWriter.hpp

```diff
@@ -120,15 +131,37 @@ class RootDigitizationWriter final : public WriterT<MeasurementContainer> {
       setupTree(treeName);
     }
 
+    /// Non-trivial destructor for memory cleanup
+    ~DigitizationTree() {
+      delete chLoc0;
+      delete chLoc1;
+      delete chValue;
+    }
+
     /// Setup the dimension depended branches
     ///
-    /// @tparam parset_t Type of the parameter set
-    ///
-    /// @param ps A dummy bound parameter set
+    /// @param i the bound index in question
     void setupBoundRecBranch(Acts::BoundIndices i) {
       tree->Branch(std::string("rec_" + bNames[i]).c_str(), &recBound[i]);
     }
 
+    /// Setup the cluster related brnach
+    ///
+    /// @param i the bound index in question
+    void setupClusterBranch(Acts::BoundIndices i) {
```

Comment:
Do we call this function for `eBoundLo0` and `eBoundLoc1`, respectively? If so, the `chLoc0` etc. are set up twice?

Reply:
I changed that only the relevant branch is added to the file:
- I one-dim measurement will only have the relevant branch added